internal/radix/adaptive/node_16.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package adaptive
+
+import (
+	"bytes"
+	"sync"
+)
+
+type Node16[T any] struct {
+	partialLen  uint32
+	artNodeType uint8
```

Comment:
```suggestion
```

Seems like the `Node16` type should intrinsically know its a node16 instead of needing to store this.